pkg/resourcemonitor/types.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcemonitor
+
+import (
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+)
+
+type Args struct {
+	PodResourceSocketPath string
+	SleepInterval         time.Duration
+	Namespace             string
```

Comment:
What is the thinking behind the namespace parameter? Shouldn't we always be watching all namespaces?? Otherwise we have incomplete picture and do not do accounting correctly.

Reply:
By default the namespace is "" which means all namespaces are watched but this was made configurable for testing purposes. In cluster wheres there could be a lot of pods deployed as part of the infrastructure, this allows us to isolate and account for pods in a certain namespace. Also in PR https://github.com/kubernetes-sigs/node-feature-discovery/pull/528 we use this to account for pods deployed in the `rte` namespace.